### PR TITLE
Let material ThemeData dictate brightness if cupertinoOverrideTheme.brightness is null

### LINKED ISSUE #47255: ThemeData.brightness is not correctly propagated to Cupertino widgets
After the upgrade to Flutter 1.12.xx I discovered that the CupertinoActivityIndicator of my app turned dark.

## Steps to Reproduce

1. Have this in my `MaterialApp` :

```dart
        theme: ThemeData(
          primarySwatch: Colors.deepOrange,
          accentColor: MyColors.accentColor,
          scaffoldBackgroundColor: Color(0xFF1E1D25), // Dark color
          fontFamily: 'Roboto',
          brightness: Brightness.dark, // Dark theme
          primaryColorBrightness: Brightness.dark, // Dark brightness
          bottomSheetTheme: BottomSheetThemeData(
            backgroundColor: MyColors.transparent
          )
        )
```

With this configurations the `CupertinoActivityIndicator` still with the dark color (almost invisible in my case). Can you confirm if my `ThemeData` is well configured?

2. Show a `CupertinoActivityIndicator` in a Flutter page
3. Note that the indicator have the dark color (almost invisible)

**Target Platform:** iOS
**Devices:** iPhone 6s

## Logs
```
[√] Flutter (Channel stable, v1.12.13+hotfix.5, on Microsoft Windows [Version 10.0.18362.535], locale en-US)
    • Flutter version 1.12.13+hotfix.5 at C:\src\flutter
    • Framework revision 27321ebbad (7 days ago), 2019-12-10 18:15:01 -0800
    • Engine revision 2994f7e1e6
    • Dart version 2.7.0


[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at C:\src\android
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 28.0.3
    • ANDROID_HOME = C:\src\android
    • Java binary at: C:\Users\Pedro Massango\Downloads\android-studio-ide-192.5959023-windows\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
    • All Android licenses accepted.

[√] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[!] Android Studio (version 4.0)
    • Android Studio at C:\Users\Pedro Massango\Downloads\android-studio-ide-192.5959023-windows\android-studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

[√] IntelliJ IDEA Community Edition (version 2019.3)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2019.1.1
    • Flutter plugin version 42.1.4
    • Dart plugin version 193.5731

[√] VS Code (version 1.40.1)
    • VS Code at C:\Users\Pedro Massango\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.6.0

[√] Connected device (1 available)
    • Redmi 5 Plus • 0258ff700005 • android-arm64 • Android 8.1.0 (API 27)

! Doctor found issues in 1 category.

```

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
## Description

Make cupertino widgets interpret `ThemeData.brightness` as forcibly overriding the brightness of the subtree, if `cupertinoOverrideTheme.brightness` is null.

Fixes: https://github.com/flutter/flutter/issues/47255

## Tests

I added the following tests:

- MaterialTheme overrides the brightness

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [x] I wrote a design doc: flutter.dev/go/nullable-cupertinothemedata-brightness
   - [x] I got input from the developer relations team, specifically from: @RedBrogdon 
   - [x] I wrote a migration guide: https://github.com/flutter/website/blob/master/src/docs/release/breaking-changes/nullable-cupertinothemedata-brightness.md

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/theme_data.dart [hunk 1]
```diff
@@ -1554,7 +1554,7 @@ class MaterialBasedCupertinoThemeData extends CupertinoThemeData {
       // like barBackgroundColor can still behave like a normal
       // CupertinoThemeData.
       super.raw(
-        _cupertinoOverrideTheme.brightness,
+        _cupertinoOverrideTheme.brightness ?? _materialTheme.brightness,
```
Comment (darrenaustin) [inline packages/flutter/lib/src/material/theme_data.dart]: This looks reasonable, but why aren't the other properties here defaulting to what is in the `_materialTheme` as well? Some may not have a reasonable default, but it seems weird to only do this for brightness.

Comment (darrenaustin) [reply packages/flutter/lib/src/material/theme_data.dart]: We should document this behavior somewhere (probably in the doc string for the `MaterialBasedCupertinoThemeData` class.

**Review by darrenaustin — COMMENTED** [review]
Looks good. Just needs some doc, and I would be curious why we aren't doing the same thing for the other properties in the override theme.

File: packages/flutter/lib/src/material/theme_data.dart (hunk 1 above)
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/material/theme_data.dart]: I think `MaterialBasedCupertinoThemeData` wants to differentiate the part of the theme that comes from `_cupertinoOverrideTheme` from the rest. From `MaterialBasedCupertinoThemeData.copyWith`'s API doc:

>   /// Copies the [ThemeData]'s `cupertinoOverrideTheme`.
  ///
  /// Only the specified override attributes of the [ThemeData]'s
  /// `cupertinoOverrideTheme` and the newly specified parameters are in the
  /// returned [CupertinoThemeData]. No derived attributes from iOS defaults or
  /// from cascaded Material theme attributes are copied.
  ///
  /// [MaterialBasedCupertinoThemeData.copyWith] cannot change the base
  /// Material [ThemeData]. To change the base Material [ThemeData], create a
  /// new Material [Theme] and use `copyWith` on the Material [ThemeData]
  /// instead.

**Comment by LongCatIsLooong** [comment]
@darrenaustin @HansMuller changed the approach. Ready for review.

**Comment by LongCatIsLooong** [comment]
@darrenaustin @HansMuller could you take another look? Made a few minor documentation updates.

**Review by darrenaustin — APPROVED** [review]
LGTM.

**Pull request merged**